Cpp/Library/Control/Controller.hh

```diff
@@ -76,43 +88,43 @@ namespace Nymph
             MEMVAR_REF_MUTABLE( std::condition_variable, CondVarContinue );
             MEMVAR_REF_MUTABLE( std::condition_variable, CondVarBreak );
             MEMVAR_NOSET( bool, BreakFlag );
-            //MEMVAR_SHARED_PTR_CONST( ReturnBufferBase, ReturnPtr );
 
         protected:
             void do_cancellation( int code );
 
+        public:
+            /// Checks whether the return buffer has been filled
+            bool HasReturn() const;
+
+            /// Get the return buffer
+            template< typename... XArgs >
+            std::tuple< XArgs&... >& GetReturn();
+
+        protected:
+            std::unique_ptr< ReturnBufferBase > fReturnBuffer;
+            MEMVAR_REF_MUTABLE( std::mutex, ReturnMutex );
+
     };
 
-/*
-    template< typename... Args >
-    void SharedControl::Break( Args&... args )
+    inline bool Controller::HasReturn() const
     {
-        while( IsAtBreak() && ! IsCanceled() )
-        {
-            if( ! WaitToContinue() )
-            {
-                THROW_EXCEPT_HERE( Exception() << "Canceled while waiting to initiate a breakpoint" );
-            }
-        }
-
-        std::unique_lock< std::mutex > lock( fMutex );
-        fBreakFlag = true;
-        fCondVarBreak.notify_all();
-        return;
+        return fReturnBuffer.operator bool();
     }
 
+    template< typename... XArgs >
+    void Controller::BreakAndReturn( XArgs&... args )
+    {
+        this->Break();
+        fReturnBuffer.reset( new ReturnBuffer( args... ) );
```

Comment:
This line does not compile it needs to be
`fReturnBuffer.reset( new ReturnBuffer<XArgs...>( args... ) );` 
But I suggest to change it to `fReturnBuffer = std::make_unique<ReturnBuffer<XArgs...>>( args... );`

Reply:
Agreed